cmd/contour/serve.go

```diff
@@ -684,29 +687,84 @@ func (s *Server) doServe() error {
 		return err
 	}
 
-	// Register an informer to watch envoy's service if we haven't been given static details.
-	if lbAddress := contourConfiguration.Ingress.StatusAddress; len(lbAddress) > 0 {
-		s.log.WithField("loadbalancer-address", lbAddress).Info("Using supplied information for Ingress status")
-		lbsw.lbStatus <- parseStatusFlag(lbAddress)
-	} else {
-		serviceHandler := &k8s.ServiceStatusLoadBalancerWatcher{
-			ServiceName: contourConfiguration.Envoy.Service.Name,
-			LBStatus:    lbsw.lbStatus,
-			Log:         s.log.WithField("context", "serviceStatusLoadBalancerWatcher"),
+	if contourConfiguration.Envoy.LoadBalancer != "" {
```

Comment:
When user does not create `ContourConfig` CR, this will still never be empty string: It gets unconditionally set [here](https://github.com/hligit/contour/blob/98835b0e0a3a397eadfcb5096ff41cefc05fbcd0/cmd/contour/servecontext.go#L580) with [this](https://github.com/hligit/contour/blob/98835b0e0a3a397eadfcb5096ff41cefc05fbcd0/pkg/config/parameters.go#L1034) value via [this](https://github.com/hligit/contour/blob/98835b0e0a3a397eadfcb5096ff41cefc05fbcd0/cmd/contour/serve.go#L349-L352) execution path which fills in the defaults. As result, old options never take effect, even when the new option was not explicitly set.

Reply:
Good catch! Thanks! I have removed setting value in `Defaults()`.
Also changed the order of flags by inspecting if ingress-status-address is empty, than the new flag and lastly the `envoy-service-name` and `envoy-service-namespace`, because they always have [default values set](https://github.com/hligit/contour/blob/a216475ca1000ad26659de09a64edff4090b8bc2/pkg/config/parameters.go#L1028-L1029).

Do you think this is good enough or we should check if the envoy service flags are explicitly set and should take precedence of the new flags?